cterasdk/core/files/collaboration.py

```diff
@@ -0,0 +1,156 @@
+import logging
+from . import common
+from ..enum import ProtectionLevel, CollaboratorType, SearchType, PortalAccountType, FileAccessMode
+from ..types import UserAccount, GroupAccount
+from ...common import Object
+from ...exception import CTERAException
+
+
+def _search_collaboration_member(ctera_host, account, cloud_folder_uid):
+    search_param = Object()
+
+    if account.account_type == PortalAccountType.User:
+        search_param.searchType = SearchType.Users
+    elif account.account_type == PortalAccountType.Group:
+        search_param.searchType = SearchType.Groups
+    else:
+        raise CTERAException("Invalid account type", None, account_type=account.account_type)
+
+    search_param.searchTerm = account.name
+    search_param.resourceUid = cloud_folder_uid
+    search_param.countLimit = 1
+    response = ctera_host.execute('', 'searchCollaborationMembers', search_param)
+    if not response.objects:
+        logging.getLogger().warning('Could not find results that match your search criteria. %s', {'name': account.name})
+    elif len(response.objects) > 1:
+        logging.getLogger().warning('Search returned multiple results. Skipping...')
+    else:
+        collaborator = response.objects.pop()
+        if account.is_local and account.name == collaborator.name:
+            logging.getLogger().debug('Found local account. %s', {'name': account.name})
+            return collaborator
+        if not account.is_local and account.directory == collaborator.domain and account.name == collaborator.name:
+            logging.getLogger().debug('Found domain account. %s', {'name': account.name, 'domain': account.directory})
+            return collaborator
+    return None
+
+
+def share(ctera_host, path, recipients, as_project, allow_reshare, allow_sync):
+    resource_info = common.get_resource_info(ctera_host, path)
+    cloud_folder_uid = resource_info.cloudFolderInfo.uid
+    if len(path.relativepath.parts) > 1:  # The shared path is not a cloud folder. Therefore, the following attrs aren't customizable.
+        as_project = False
+        allow_sync = False
+
+    valid_recipients = []
+    for recipient in recipients:
+        if _valid_recipient(recipient):
+            if recipient.type in [CollaboratorType.LU, CollaboratorType.LG, CollaboratorType.DU, CollaboratorType.DG]:
+                collaborator = _search_collaboration_member(ctera_host, recipient.account, cloud_folder_uid)
+                if collaborator:
+                    recipient.collaborator = collaborator
+            valid_recipients.append(recipient)
+
+    if valid_recipients:
+        share_param = _create_share_param(path.fullpath(), as_project, allow_reshare, allow_sync)
+        for recipient in valid_recipients:
+            _add_share_recipient(share_param, recipient)
+
+        members = [str(recipient) for recipient in valid_recipients]
```

Comment:
You can move the array to the print. This way you first avoid the additional variable and also if the log level if above info, the array will not be rendered at all

Reply:
Fixed. I did that initially, but the log line was too long > 140 chars.